app/workers/detect_invariants.rb

```diff
@@ -185,6 +186,30 @@ def detect_application_choices_with_courses_from_the_incorrect_cycle
     end
   end
 
+  def detect_submitted_applications_with_an_incorrect_number_of_selected_references
+    applications_with_incorrect_number_of_references = ApplicationChoice
+    .joins(application_form: [:application_references])
+    .where.not(application_choices: { status: 'unsubmitted' })
+    .where(references: { feedback_status: :feedback_provided })
+    .where(references: { selected: true })
+    .group('id')
+    .having('COUNT("references".id) != 2')
```

Comment:
The data migration related to this PR fixes counts greater than 2 but doesn't touch those applications with less than 2 selected references. There are 5 submitted application forms in the prod DB with less than 1 selected reference.